spec/index.bs

```diff
@@ -9757,11 +9742,27 @@ interface GPUCanvasContext {
             **Returns:** {{GPUTexture}}
 
             1. If |this|.{{GPUCanvasContext/[[configuration]]}} is `null`:
-                1. Throw an {{OperationError}} and stop.
+                1. Throw an {{InvalidStateError}} and stop.
+            1. [=Assert=] |this|.{{GPUCanvasContext/[[textureDescriptor]]}} is not `null`.
+            1. Let |device| be |this|.{{GPUCanvasContext/[[configuration]]}}.{{GPUCanvasConfiguration/device}}.
             1. If |this|.{{GPUCanvasContext/[[currentTexture]]}} is `null` or
                 |this|.{{GPUCanvasContext/[[currentTexture]]}}.{{GPUTexture/[[destroyed]]}} is true:
-                1. Set |this|.{{GPUCanvasContext/[[currentTexture]]}} to the result of [$allocating
-                    a new context texture$] for |this|.
+                1. Set |this|.{{GPUCanvasContext/[[currentTexture]]}} to the result of calling
+                    |device|.{{GPUDevice/createTexture()}} with |this|.{{GPUCanvasContext/[[textureDescriptor]]}}.
+
+                    Note: If the texture can't be created (e.g. due to validation failure or out-of-memory),
+                    this generates and error and returns an [=invalid=] {{GPUTexture}}.
+                    This is true even if the same validation already failed in {{GPUCanvasContext/configure()}}.
+                    Implementations may deduplicate this validation, but **must** still resurface the validation error here.
+
+                    Note: If a previously presented texture from |this| matches the required criteria,
+                    its GPU memory may be re-used. This optimization is unobservable; the returned texture is always cleared.
+
+                    Issue: FIXME: If this fails, can you attempt to getCurrentTexture again (e.g. after freeing some resources)?
+                    If yes, how does getCurrentTexture know to try again instead of returning the same failed texture?
+                    It's not possible to check `valid` from the content timeline.
+                    There's also a related issue here that `destroyed` is checked from the content timeline (technically possible
+                    but not intended).
```

Comment:
note to self: I need to figure this out before unmarking as draft

Reply:
never mind, it doesn't matter because you can't reliably know that it failed in the same frame anyway.